libarchive/archive_read_support_format_iso9660.c

```diff
@@ -882,21 +887,43 @@ isPVD(struct iso9660 *iso9660, const unsigned char *h)
 	if (!isNull(iso9660, h, PVD_reserved2_offset, PVD_reserved2_size))
 		return (0);
 
+	/* Volume space size must be encoded according to 7.3.3 */
+	if (!isValid733Integer(h + PVD_volume_space_size_offset)) {
```

Comment:
A lot of fields in the PVD use section 7.2.3 or 7.3.3 encoding (which stores two copies of the integer).  We can compare those two copies to help verify that this really is an ISO9660 image.

Important:  We really want to reject anything that's not an ISO9660 image.  So we want to be as strict as possible, but not so strict that we reject valid ISO9660 images that we can support.  It's a tricky balance.